src/audio/eq_fir/eq_fir.c

```diff
@@ -373,6 +373,78 @@ static int eq_fir_setup(struct comp_dev *dev, struct comp_data *cd, int nch)
 	return 0;
 }
 
+static int eq_fir_validator(struct comp_dev *dev, void *new_data, uint32_t new_data_size)
+{
+	struct sof_eq_fir_config *config = new_data;
+	struct sof_fir_coef_data *eq;
+	int16_t *assign_response;
+	int16_t *coef_data;
+	int i;
+	int j;
+	int nch = -EINVAL;
+	struct list_item *blist;
+
+	comp_err(dev, "eq_fir_validator()");
+
+	/* Fetch the number of channels */
+	list_for_item(blist, &dev->bsource_list) {
```

Comment:
I suppose during the component initialization I could store that info in cd, yeah. Just need to figure out how the rest of the component does this.

This function to be fair doesn't run inside the processing loop (and the channel search does, in the module_adapter). I'll consider doing this though.

Reply:
Though sending new blobs many times on the fly is not normal usage scenario so the overhead is likely not much an issue. I have in my to-do list the addition of enum control to change the response assign vector on the fly with enum controls. Then e.g. device orientation or mode based EQ switch would be handled with quick enum control instead. All needed responses can be packed to a single blob. We did  not have enum control support in FW when I did this so the feature has not been done yet.

Reply:
Yeah this overhead is specific to EQ FIR, rather than common to _any_ component. Each component can make its own kind of verification, including one that doesn't need to look at the input buffer(s) in the first place. The FIR EQ specific part both solves the original problem AND motivates the generic data_blob stuff.